amazon-kinesis-client/src/main/java/software/amazon/kinesis/coordinator/ExecutorStateEvent.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package software.amazon.kinesis.coordinator;
+
+import lombok.Getter;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import software.amazon.kinesis.annotations.KinesisClientInternalApi;
+import software.amazon.kinesis.leases.LeaseCoordinator;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+
+@Getter
+@ToString(exclude = "MESSAGE")
+@Slf4j
+@KinesisClientInternalApi
+public class ExecutorStateEvent implements DiagnosticEvent {
+    private final String MESSAGE = "Current thread pool executor state: ";
+
+    private int currentQueueSize;
+    private int activeThreads;
+    private int coreThreads;
+    private int leasesOwned;
+    private int largestPoolSize;
+    private int maximumPoolSize;
+
+    public ExecutorStateEvent(ExecutorService executor, LeaseCoordinator leaseCoordinator) {
+        if (executor instanceof ThreadPoolExecutor) {
```

Comment:
this method seem to work only for ThreadPoolExecutor, can we take the specific type in the arguments?

Reply:
We can do that. Because the Scheduler's executor service member is typed more generally as ExecutorService, we'll get a compilation error unless we cast it when we create the ExecutorStateEvent, like so:

`new ExecutorStateEvent((ThreadPoolExecutor) executorService, leaseCoordinator);`

Looks a little less pretty to me but I agree that it's probably better to restrict the input type rather than having a bunch of null and meaningless members

Reply:
Actually, we can't do that - as discussed force casting a generic ExecutorService into a ThreadPoolExecutor could cause crashes. We can keep the ExecutorService as the input type but only have it take meaningful action (i.e. logging) if that ExecutorService is a ThreadPoolExecutor